modules/light-clients/08-wasm/types/update.go

```diff
@@ -60,30 +51,28 @@ type (
 
 // Client state and new consensus states are updated in the store by the contract
 func (cs ClientState) UpdateState(ctx sdk.Context, cdc codec.BinaryCodec, clientStore sdk.KVStore, clientMsg exported.ClientMessage) []exported.Height {
-	header, ok := clientMsg.(*Header)
+	clientMessage, ok := clientMsg.(*ClientMessage)
 	if !ok {
-		panic(fmt.Errorf("expected type %T, got %T", &Header{}, clientMsg))
+		panic(fmt.Errorf("expected type %T, got %T", &ClientMessage{}, clientMsg))
```

Comment:
if we're going to panic here, then we should also panic in `CheckForMisbehaviour` and `UpdateStateOnMisbehaviour`  if `clientMsg` is not `*ClientMessage`, right?

Reply:
I think this mostly followed what is done for tendermint. For CheckForMisbehaviour, returning `false` would call `UpdateState` and result in that panic being hit. I guess panicking could be done here but it wouldn't save us much right?

Reply:
Agree!